test/edu/washington/escience/myria/operator/SplitTest.java

```diff
@@ -0,0 +1,167 @@
+    TupleBatch result;
+    int rowIdx = 0;
+    while (!splitOp.eos()) {
+      result = splitOp.nextReady();
+      if (result != null) {
+        assertEquals(schema.numColumns(), result.getSchema().numColumns());
+        assertEquals(Type.BOOLEAN_TYPE, result.getSchema().getColumnType(0));
+        assertEquals(Type.STRING_TYPE, result.getSchema().getColumnType(1));
+        assertEquals(Type.LONG_TYPE, result.getSchema().getColumnType(2));
+        assertEquals(Type.DOUBLE_TYPE, result.getSchema().getColumnType(3));
+
+        for (int batchIdx = 0; batchIdx < result.numTuples(); ++batchIdx, ++rowIdx) {
+          assertEquals(((Boolean) expectedResults[rowIdx][0]).booleanValue(),
+              result.getBoolean(0, batchIdx));
+          assertEquals((expectedResults[rowIdx][1]).toString(), result.getString(1, batchIdx));
+          assertEquals(((Long) expectedResults[rowIdx][2]).longValue(), result.getLong(2, batchIdx));
+          assertEquals(
+              Double.doubleToLongBits(((Double) expectedResults[rowIdx][3]).doubleValue()),
+              Double.doubleToLongBits(result.getDouble(3, batchIdx)));
+
+        }
+      }
+    }
+    assertEquals(expectedResults.length, rowIdx);
+    splitOp.close();
+  }
+
+  @Test
+  public void testGeneratedSplitsSingleColumn() throws DbException {
+    final String[] expectedResults = {"foo", "bar", "baz"};
+    final Schema schema =
+        new Schema(ImmutableList.of(Type.STRING_TYPE), ImmutableList.of("string"));
+    final TupleBatchBuffer input = new TupleBatchBuffer(schema);
+    input.putString(0, "foo:bar:baz");
+    Split splitOp = new Split(new TupleSource(input), 0, ":");
+
+    splitOp.open(TestEnvVars.get());
+    TupleBatch result;
+    int rowIdx = 0;
+    while (!splitOp.eos()) {
+      result = splitOp.nextReady();
+      if (result != null) {
+        assertEquals(schema.numColumns(), result.getSchema().numColumns());
+        assertEquals(Type.STRING_TYPE, result.getSchema().getColumnType(0));
+
+        for (int batchIdx = 0; batchIdx < result.numTuples(); ++batchIdx, ++rowIdx) {
+          assertEquals(expectedResults[rowIdx], result.getString(0, batchIdx));
+        }
+      }
+    }
+    assertEquals(expectedResults.length, rowIdx);
+    splitOp.close();
+  }
+
+  /**
+   * Test output spanning multiple batches. All integers from 0 to 2 * TupleBatch.BATCH_SIZE + 1 are
+   * concatenated as comma-separated strings in rows of 10 each. Result should contain each integer
+   * from the input in its own row.
+   * 
+   * @throws DbException
+   */
+  @Test
+  public void testAllBatchesReturned() throws DbException {
+
+    final Schema schema =
+        new Schema(ImmutableList.of(Type.STRING_TYPE), ImmutableList.of("joined_ints"));
+    final TupleBatchBuffer input = new TupleBatchBuffer(schema);
+    final long expectedResults = 2 * TupleBatch.BATCH_SIZE + 1;
+    final int intsInRow = 10;
+    for (long i = 0; i < (expectedResults / intsInRow) + 1; ++i) {
```

Comment:
I was concerned I would run into a row size limit in Myria (this would be 20001 ints in one row, a few bytes each in ASCII, with an extra byte for each comma).